core/chain-configs/src/genesis_config.rs

```diff
@@ -175,11 +177,29 @@ impl From<&GenesisConfig> for EpochConfig {
 )]
 pub struct GenesisRecords(pub Vec<StateRecord>);
 
+#[derive(Debug, Copy, Clone)]
+pub enum GenesisRecordsFileType {
+    FullGenesis,
+    RecordsArray,
+}
+
+#[derive(Debug, Clone)]
+pub struct GenesisRecordsFile {
+    pub path: PathBuf,
+    pub file_type: GenesisRecordsFileType,
+}
+
 #[derive(Debug, Clone, Default, Serialize, Deserialize)]
 pub struct Genesis {
     #[serde(flatten)]
     pub config: GenesisConfig,
     pub records: GenesisRecords,
+    /// Genesis object may not contain records
+    /// In this case records can be found in records_file
+    /// The idea is that all records consume too much memory
+    /// So they should be processed in streaming fashion with stream_records_with_callback
```

Comment:
Nit: let's add `.` to the sentences. I don't think proper grammar and punctuation are useful in and of themselves, however, (at least for myself), if you try to not just "add comments" but write proper sentences, the end result is better. This is basically a trick to make yourself be more thoughtful when translating your internal understanding of the problem to another human, who might lack the context.